pkg/minikube/constants/constants.go

```diff
@@ -62,6 +67,45 @@ func getMinishiftHomeDir() string {
 	homeEnv, ok := os.LookupEnv(MiniShiftHomeEnv)
 	if ok {
 		return homeEnv
+	} else if IsProfileExists() {
+		return filepath.Join(util.HomeDir(), ".minishift", GetMachineName())
+	} else {
+		return filepath.Join(util.HomeDir(), ".minishift")
+	}
+}
+
+//Checks viper or allinstanceconfg to say if profile is defined.
+func IsProfileExists() bool {
+	if viper.GetString("profile") != "" {
+		return true
+	} else if minishiftConfig.AllInstancesConfig != nil {
+		if profileActions.GetActiveProfile() != "" {
+			return true
+		}
+	}
+	return false
+}
+
+//Returns the machines after taking in to account if --profile or
+//profile information available in allinstance config
+func GetMachineName() string {
+	activeProfile := profileActions.GetActiveProfile()
+	if viper.GetString("profile") != "" {
+		return viper.GetString("profile")
+	} else if activeProfile != "" {
+		return activeProfile
+	} else {
+		return DefaultMachineName
+	}
+}
+
+//Returns Minishift instance path. Its considers the profile if defined and used
+func GetMinipath() string {
+	homeEnv, ok := os.LookupEnv(MiniShiftHomeEnv)
+	if ok {
+		return homeEnv
+	} else if IsProfileExists() {
+		return filepath.Join(util.HomeDir(), ".minishift", GetMachineName())
```

Comment:
BTW, you might be better off making this func GetMinipath(profile string) string. This decouples GetMiniPath from getMachineName. As a side effect it becomes easier to test. 

Developing with testing in mind often leads to a better, less coupled design.